packages/nimble-components/src/rich-text-editor/tests/rich-text-editor.spec.ts

```diff
@@ -1302,6 +1304,160 @@ describe('RichTextEditor', () => {
+            const editor = element.shadowRoot?.querySelector('.editor');
+            expect(editor!.getAttribute('aria-disabled')).toBe('false');
+
+            await pageObject.setDisabledState();
+
+            expect(editor!.getAttribute('aria-disabled')).toBe('true');
+        });
+
+        it('should reflect disabled value to the "contenteditable" attribute of tiptap editor', async () => {
+            const editor = element.shadowRoot?.querySelector('.ProseMirror');
+            expect(editor!.getAttribute('contenteditable')).toBe('true');
+
+            await pageObject.setDisabledState();
+
+            expect(editor!.getAttribute('contenteditable')).toBe('false');
+        });
+
+        it('should change the tabindex value of the editor when disabled value changes', async () => {
+            expect(pageObject.getEditorTabIndex()).toBe('0');
+
+            await pageObject.setDisabledState();
+
+            expect(pageObject.getEditorTabIndex()).toBe('-1');
+        });
+
+        describe('should reflect disabled value to the disabled and aria-disabled state of toggle buttons', () => {
+            const focused: string[] = [];
+            const disabled: string[] = [];
+            for (const value of formattingButtons) {
+                const specType = getSpecTypeByNamedList(
+                    value,
+                    focused,
+                    disabled
+                );
+                specType(
+                    `for "${value.name}" button`,
+                    // eslint-disable-next-line @typescript-eslint/no-loop-func
+                    async () => {
+                        expect(
+                            pageObject.isButtonDisabled(
+                                value.toolbarButtonIndex
+                            )
+                        ).toBeFalse();
+
+                        await pageObject.setDisabledState();
+
+                        expect(
+                            pageObject.isButtonDisabled(
+                                value.toolbarButtonIndex
+                            )
+                        ).toBeTrue();
+                    }
+                );
+            }
+        });
+    });
+
+    it('should hide the footer when "footer-hidden" attribute is enabled', async () => {
+        expect(pageObject.isFooterHidden()).toBeFalse();
+
+        await pageObject.hideFooter();
+
+        expect(pageObject.isFooterHidden()).toBeTrue();
+    });
+
+    it('should fire "input" event when there is an input to the editor', async () => {
+        const inputEventListener = createEventListener(element, 'input');
+
+        await pageObject.setEditorTextContent('input');
+        await inputEventListener.promise;
+
+        expect(inputEventListener.spy).toHaveBeenCalledTimes(1);
+    });
+
+    it('should fire "input" event when setting the content through "setMarkdown"', async () => {
```

Comment:
Is there a concrete need to fire the "input" event when the content is set programmatically? As a general rule we've tried to only fire events on interactive input, not programmatic. Usually applications are aware of the programmatic changes (since they're the ones making them) so they don't need to be notified of them. And often apps don't want to be notified of them, since it can cause unintended side effects.

Reply:
It makes sense not to fire the `input` event when they set content programmatically. So I removed the firing input event when setting the markdown content programmatically.